packages/utils/src/shorthand.ts

```diff
@@ -0,0 +1,610 @@
+  ],
+  'scroll-padding': [
+    'scroll-padding-block',
+    'scroll-padding-block-end',
+    'scroll-padding-block-start',
+    'scroll-padding-bottom',
+    'scroll-padding-inline',
+    'scroll-padding-inline-end',
+    'scroll-padding-inline-start',
+    'scroll-padding-left',
+    'scroll-padding-right',
+    'scroll-padding-top',
+  ],
+  'scroll-padding-block': [
+    'scroll-padding-block-start',
+    'scroll-padding-block-end',
+    'scroll-padding-top',
+    'scroll-padding-bottom',
+  ],
+  'scroll-padding-inline': [
+    'scroll-padding-inline-start',
+    'scroll-padding-inline-end',
+    'scroll-padding-left',
+    'scroll-padding-right',
+  ],
+  'scroll-timeline': ['scroll-timeline-name', 'scroll-timeline-axis'],
+  'text-decoration': [
+    'text-decoration-color',
+    'text-decoration-line',
+    'text-decoration-style',
+    'text-decoration-thickness',
+  ],
+  'text-emphasis': ['text-emphasis-color', 'text-emphasis-style'],
+  'text-wrap': ['text-wrap-mode', 'text-wrap-style'],
+  transition: [
+    'transition-behavior',
+    'transition-delay',
+    'transition-duration',
+    'transition-property',
+    'transition-timing-function',
+  ],
+  'view-timeline': ['view-timeline-name', 'view-timeline-axis'],
+} as const;
+
+// Make sure to update the `shorthandBuckets` variable in `packages/react/src/runtime/shorthand.ts`
+// after making changes to this object.
+export const shorthandBuckets = {
+  all: 0,
+  animation: 1,
+  'animation-range': 1,
+  background: 1,
+
+  border: 1,
+  'border-color': 2,
+  'border-style': 2,
+  'border-width': 2,
+
+  'border-block': 3,
+  'border-inline': 3,
+
+  'border-top': 4,
+  'border-right': 4,
+  'border-bottom': 4,
+  'border-left': 4,
+
+  'border-block-start': 5,
+  'border-block-end': 5,
+  'border-inline-start': 5,
+  'border-inline-end': 5,
+
+  'border-image': 1,
+  'border-radius': 1,
+
+  'column-rule': 1,
+  columns: 1,
+  'contain-intrinsic-size': 1,
+  container: 1,
+  flex: 1,
+  'flex-flow': 1,
+  font: 1,
+  'font-synthesis': 1,
+  'font-variant': 2,
+  gap: 1,
+  grid: 1,
+  'grid-area': 1,
+  'grid-column': 2,
+  'grid-row': 2,
+  'grid-template': 2,
+  inset: 1,
+  'inset-block': 2,
+  'inset-inline': 2,
+  'list-style': 1,
+
+  margin: 1,
+  'margin-block': 2,
+  'margin-inline': 2,
+
+  mask: 1,
+  'mask-border': 1,
+  offset: 1,
+  outline: 1,
+  overflow: 1,
+  'overscroll-behavior': 1,
+
+  padding: 1,
+  'padding-block': 2,
+  'padding-inline': 2,
+
+  'place-content': 1,
+  'place-items': 1,
+  'place-self': 1,
+  'position-try': 1,
+
+  'scroll-margin': 1,
+  'scroll-margin-block': 2,
+  'scroll-margin-inline': 2,
+
+  'scroll-padding': 1,
+  'scroll-padding-block': 2,
+  'scroll-padding-inline': 2,
+
+  'scroll-timeline': 1,
+  'text-decoration': 1,
+  'text-emphasis': 1,
+  'text-wrap': 1,
+  transition: 1,
+  'view-timeline': 1,
+} as const satisfies Record<ShorthandProperties, Depths>;
+// ^^ this type lets us enforce that the copy of shorthandBuckets
```

Comment:
To question aloud, _why_ do we even have this in this file?

…what if we imported this into the other file, doesn't that just work?  I'd expect `shorthandFor` to be tree shaken, am I wrong?

It's fine, it feels maintainable, just looking at it again, I'm not sure why it's in both places…

Reply:
ah i made two copies of `shorthandBuckets` because in `@compiled/react/runtime`, importing `@compiled/utils` bloats the bundle size and causes `npx size-limit` to fail yeah

re: tree shaking, can we assume that tree-shaking always runs in realistic Compiled usage? if so, i can see if i can update the `size-limit` config to account for tree-shaking maybe

Reply:
I have no clue 😸